src/main/java/uk/gov/hmcts/opal/util/HttpUtil.java

```diff
@@ -41,4 +45,23 @@ public static String extractPreferredUsername(HttpServletRequest request, Access
             .map(tokenService::extractPreferredUsername)
             .orElse(null);
     }
+
+    public static Role getRequiredRole(UserState userState, short businessUnitId) {
+        return userState.getRoleForBusinessUnit(businessUnitId).orElseThrow(() -> new
+            AccessDeniedException("User does not have an assigned role in business unit: " + businessUnitId));
+    }
+
+    public static boolean checkRoleHasPermission(Role role, Permissions permission) {
+        if (role.doesNotHavePermission(permission)) {
+            throw new AccessDeniedException("User does not have the required permission: " + permission.description);
+        }
+        return true;
+    }
+
+    public static boolean checkAnyRoleHasPermission(UserState userState, Permissions permission) {
+        if (userState.noRoleHasPermission(permission)) {
+            throw new AccessDeniedException("User does not have the required permission: " + permission.description);
+        }
+        return true;
+    }
```

Comment:
These methods can be moved in it's own utility class (i.e. RoleUtils or PermissionUtil etc.) rather than be in HttpUtil.

Reply:
Moved to a PermissionUtil class